keyboards/splitkb/kyria/keymaps/squigglybob/keymap.c

```diff
@@ -0,0 +1,315 @@
+/* Copyright 2019 Thomas Baart<thomas @splitkb.com> **This program is free software : you can redistribute it and / or modify *it under the terms of the GNU General Public License as published by *the Free Software Foundation, either version 2 of the License, or *(at your option)any later version.**This program is distributed in the hope that it will be useful, *but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum layers {
+    _DVERTY = 0,
+    _QWERTY,
+    _DVORAK,
+    _NAV,
+    _SYM,
+    _SYM_DVERTY,
+    _FUNCTION,
+    _ADJUST,
+};
+
+// Aliases for readability
+#define QWERTY DF(_QWERTY)
+#define DVERTY DF(_DVERTY)
+#define DVORAK DF(_DVORAK)
```

Comment:
If you want these persistent (eg, after powering down your system, etc), there is the `set_single_persistent_default_layer(layer)` function which requires custom keycodes.  Or you can use something like this:
```c
#if defined(AUDIO_ENABLE) && defined(DEFAULT_LAYER_SONGS)
static float default_layer_songs[][MAX_LAYER][2] = DEFAULT_LAYER_SONGS;
#endif

layer_state_t default_layer_state_set_user(layer_state_t state) {
    static bool has_init_been_ran = false;
    // We don't want to run this the first time it's called, since it's read from eeeprom and called
    // as part of the startup process. But after that, it's okay.
    if (has_init_been_ran) {
#if defined(AUDIO_ENABLE) && defined(DEFAULT_LAYER_SONGS)
        if (get_highest_layer(state) < MAX_LAYER) {
            PLAY_SONG(default_layer_songs[get_highest_layer(state)]);
        }
#endif
        eeconfig_update_default_layer(state);
    }
    has_init_been_ran = true;

    return state;
}
```
This skips the first run after start, but will write to eeprom after that, for each time the DF(x) keycodes are used.

Reply:
I don't think I've had any issues with this bit of code as far as I can tell.
The buttons all do as they're programmed to do every time the kyria turns on.
Does it need changing?

Reply:
Well, wasn't sure of the use case here, and figured it was info worth posting. But also, no action needed.